rust/datafusion/README.md

```diff
@@ -58,6 +58,49 @@ Here are some of the projects known to use DataFusion:
 
 (if you know of another project, please submit a PR to add a link!)
 
+## Example Usage
+
+Run a SQL query against data stored in a CSV:
+
+```rust
+  let mut ctx = ExecutionContext::new();
+  ctx.register_csv("example", "tests/example.csv", CsvReadOptions::new())?;
+
+  // Create a plan to run a SQL query
+  let df = ctx.sql("SELECT a, MIN(b) FROM example GROUP BY a LIMIT 100")?;
+
+  // execute and print results
+  let results: Vec<RecordBatch> = df.collect().await?;
+  print_batches(&results)?;
+```
+
+Use the DataFrame API to process data stored in a CSV:
+
+```rust
```

Comment:
Nitpicking: this might be a little bit fun with API churn, e.g. I believe the input expr ownership work you've recently opened would change these from slices to vecs and we don't have a way to catch that automatically like we do for the in-crate docs (am I right in thinking that `cargo test` runs all doctests?).

Reply:
This is a good point @returnString  

The way I justified the danger of drift to myself was "the main usecase of this documentation (the overview) is likely to help them answer the question of "should I even bother to try and use this crate". Once they decide to try and actually use the crate they will look at the real docs on docs.rs (from which they can copy/paste).

For the purpose of an example of "what does this library do" I felt even a slightly out of date example might be valuable.

Or maybe I am just trying to pad my github stats ;) But in all seriousness I am not committed to this PR. If it isn't a good idea I can just close it

Reply:
That makes sense to me; agreed that (personally, at least) I'll give less consideration to projects without simple readme examples.

It balloons the scope of this PR quite a lot so I'm not saying this is a _good_ idea, but I just did a bit of digging and it looks like people have gone through this particular problem before: https://blog.guillaume-gomez.fr/articles/2019-04-13+Keeping+Rust+projects%27+README.md+code+examples+up-to-date

And the end result of that  is https://crates.io/crates/doc-comment, which looks like it'll wire up any `rust`-tagged code blocks in external files as doctests, optionally only for `#[cfg(test)]`.

If it's useful, I could log a followup task to integrate that and take a look at it myself?

Reply:
@returnString  https://crates.io/crates/doc-comment looks super awesome -- I think that would be most helpful

Reply:
Logged as https://issues.apache.org/jira/browse/ARROW-12015 :)